app/models/concerns/key_pair_compute_resource.rb

```diff
@@ -23,8 +23,12 @@ def recreate
     setup_key_pair
   end
 
-  def delete_key_pair(kay_pair_name)
-    delete_key_from_resource(kay_pair_name)
+  def delete_key_from_resource(remote_key_pair = key_pair)
+    logger.info "removing #{provider_friendly_name}: #{name} key #{remote_key_pair.name}"
+    key = client.key_pairs.get(remote_key_pair.name)
+    key.nil? || key.destroy
```

Comment:
Nitpicks: This could be just `client.key_pairs.get(remote_key_pair.name).try(:destroy)`. Also the info log above is a little bit confusing when rendered as `removing openstack: myopenstack key foreman-XYZ". There might be a better wording.